File: Semester_3/DSAL/lab2/q1.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>

void Reverse(int n, int a[100], int *p)
{
	int i;
	p = &a[n-1];
	printf("The elements of the array in reverse order are : ");
	for(i =0; i<n; i++)
		printf("%d ", *p--);
}


int main()
{
	int i, n, *p, a[100];
	printf("Enter the number of integers of the array: \n");
	scanf("%d", &n);

	printf("Enter %d elements of the array : ", n);
	for(i =0; i<n; i++)
	{
		scanf("%d", &a[i]);
	}
	Reverse(n, a, p);
	return 0;
}

File: Semester_3/OOP/week2/q1.java
import java.util.Scanner;

class q1
{
	static void asc(int arr1[], int n)
	{
		int i, j; 
		for(i =0; i<n; i++)
		{
			for(j =0; j<n-i-1; j++)
			{
				if(arr1[j]>arr1[j+1])
				{
					arr1[j] = arr1[j] + arr1[j+1];
					arr1[j+1] = arr1[j] - arr1[j+1];
					arr1[j] = arr1[j] - arr1[j+1];
				}
			}
		}
		System.out.println("The array in ascending order is : \n");
		for(i = 0; i<arr1.length; i++)
		{
			System.out.println(" " + arr1[i]);
		}
	}

	static void dsc(int arr1[], int n)
	{
		int j, i; 
		for(i = 0; i<n; i++)
		{
			for(j = 0; j<n-i-1; j++)
			{
				if (arr1[j]<arr1[j+1])
				{
					arr1[j] = arr1[j] + arr1[j+1];
					arr1[j+1] = arr1[j] - arr1[j+1];
					arr1[j] = arr1[j] - arr1[j+1];
				}
			}
		}
		System.out.println("The array in descending order is: \n");
		for(i = 0; i<arr1.length; i++)
		{
			System.out.println(" " + arr1[i]);
		}
	}
	public static void main(String[] args)
	{
		int n,i;
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter the number of elements");
		n = sc.nextInt();
		int[] arr1 = new int[n];
		System.out.println("Enter " + n + " numbers");
		for(i =0; i<n; i++)
		{
			arr1[i] = sc.nextInt();
		}
		asc(arr1, n);
		dsc(arr1, n);
	}
} 